backend.py

```diff
@@ -2,60 +2,64 @@
-    """
+def build_folders(initiator, destination_temp, standard, root, repoCluster):
+	"""Transform the repos' folder structure to that of the register
+	and build HTML pages for each standard.
+	"""
 
-    source_fs = OSFS(source)
+	source_fs = OSFS(initiator)
 
-    print "Processing %s ... " % standard['id']
-    standard_fs = source_fs.opendir(standard['id'])
+	print "Processing %s ... " % standard['id']
+    ##standard_fs = source_fs.opendir(standard['id'])
 
     # list all artifacts of a standard
-    artifacts = standard_fs.listdir(dirs_only=True)
-    if '.git' in artifacts: artifacts.remove(".git")
+	artifacts = source_fs.listdir(dirs_only=True)
+	if '.git' in artifacts: artifacts.remove(".git")
+		
 
-    for artifact in artifacts:
+	for artifact in artifacts:
         # check whether artifact folder exists in destination_temp 
-        if root.exists('%s/%s' % (destination_temp, artifact)) == False:
-            root.makedir('%s/%s' % (destination_temp, artifact))
+		if root.exists('%s/%s' % (destination_temp, artifact)) == False:
+			root.makedir('%s/%s' % (destination_temp, artifact))
 
         # copy standard folders from source to destination_temp in desired structure
-        root.copydir('%s/%s/%s' % (source, standard['id'], artifact),  '%s/%s/%s' % (destination_temp, artifact, standard['id']))
+		root.copydir('%s/%s' % (standard['id'], artifact),  '%s/%s/%s' % (destination_temp, artifact, standard['id']))
 
-    html = create_standard_webpage(standard, artifacts)
+	html = create_standard_webpage(standard, artifacts)
 
-    # check whether register/standard exists
-    if root.exists('%s/%s' % (destination_temp, standard['id'])) == False:
-        root.makedir('%s/%s' % (destination_temp, standard['id']))
+    # check whether register/cluster/standard exists
+	if root.exists('%s/%s' % (destination_temp, repoCluster)) == False:
+		root.makedir('%s/%s' % (destination_temp, repoCluster))
+	
+	if root.exists('%s/%s/%s' % (destination_temp, repoCluster, standard['id'])) == False:
+		root.makedir('%s/%s/%s' % (destination_temp, repoCluster, standard['id']))
     
     # write standard HTML page to register/standard/index.html
-    with codecs.open('%s/%s/index.html' % (destination_temp, standard['id']), 'w', encoding='utf8') as f:
-        f.write(html)
+	with codecs.open('%s/%s/%s/index.html' % (destination_temp, standard['cluster'], standard['id']), 'w', encoding='utf8') as f:
+		f.write(html)
 
     # copy web assets
-    root.copydir('web/assets', '%s/r' % destination_temp, overwrite=True)
+	root.copydir('web/assets', '%s/r' % destination_temp, overwrite=True)
 
 def fetch_repo(root, repo, url, destination_temp):
     """Clone repos from GitHub to source folder."""
 
     print "Fetching %s from %s" % (repo, url)
 
-    if root.exists('repos/%s' % repo):
+    if root.exists('%s' % repo):
         print "Repo %s exists, issuing a git pull..." % repo
-        call('cd repos/%s; git pull' % repo, shell=True)
```

Comment:
Aha maar die is nog onder constructie zie ik. ;)

2015-12-16 13:58 GMT+01:00 Simeon Nedkov <notifications@github.com>:

> In backend.py
> <https://github.com/Geonovum/technisch-register/pull/33#discussion_r47772299>
> :
>
> >          print "Repo %s exists, issuing a git pull..." % repo
> > -        call('cd repos/%s; git pull' % repo, shell=True)
>
> No worries! Ik hoop altijd dat er een briljante reden achter schuilgaat! [image:
> :relaxed:]
>
> Het maken van mapjes moet door een setup.py script geregeld worden, zie
> #30 <https://github.com/Geonovum/technisch-register/issues/30>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Geonovum/technisch-register/pull/33/files#r47772299>.
>